mesonbuild/environment.py

```diff
@@ -651,6 +653,95 @@ def __init__(self, source_dir: str, build_dir: str, options: coredata.SharedCMDO
         self.default_pkgconfig = ['pkg-config']
         self.wrap_resolver: T.Optional['Resolver'] = None
 
+    @staticmethod
+    def create_pkgconfig_repo(for_machine: MachineChoice = MachineChoice.BUILD,
+                              is_cross: bool = False,
+                              system_paths: T.Optional[T.List[str]] = None,
+                              extra_paths: T.Optional[T.List[str]] = None,
+                              sysroot_dir: T.Optional[str] = None) -> pkgconfig.Repository:
+        # FIXME: Take default from CLI:
+        # pkg-config pkg-config --variable pc_path
+        # pkg-config pkg-config --variable pc_system_libdirs
+        # pkg-config pkg-config --variable pc_system_includedirs
+        default_system_paths: T.List[Path] = []
+        default_library_paths: T.List[Path] = []
+        default_include_paths: T.List[Path] = []
+        if not is_cross or for_machine is MachineChoice.BUILD:
+            # Get some default paths for build machine
+            default_prefixes = [Path('/usr/local'), Path('/usr')]
+            default_libdir = mesonlib.default_libdir()
+            if default_libdir in {'lib', 'lib64'}:
+                default_libdirs = [default_libdir, 'share']
+            else:
+                default_libdirs = [default_libdir, 'lib', 'share']
+            for p in default_prefixes:
+                for l in default_libdirs:
+                    default_system_paths.append(p / l / 'pkgconfig')
+            default_library_paths = [Path('/usr') / l for l in default_libdirs]
+            default_include_paths = [Path('/usr/include')]
+
+        def _paths_from_env(varname: str) -> T.Optional[T.List[Path]]:
+            value = _get_env_var(for_machine, is_cross, varname)
+            if value is None:
+                return None
+            return [Path(p) for p in value.split(os.pathsep)] if value else []
+
+        # Paths where to lookup for system pc files
+        if system_paths is None:
+            system_paths = _paths_from_env('PKG_CONFIG_LIBDIR')
+            if system_paths is None:
+                system_paths = default_system_paths
+        else:
+            system_paths = [Path(p) for p in system_paths]
+
+        # Extra paths where to lookup for pc files
+        if extra_paths is None:
+            extra_paths = _paths_from_env('PKG_CONFIG_PATH')
```

Comment:
You should look into machine files as well...

Reply:
This is already being read into the `Environment` options from the environment, command line, and machine file in the correct order. I'd much prefer if we could use that instead of re-implementing it. the PKG_CONFIG_LIBDIR is much the same, but we don't currently read it at all.